api/client/accesslist/accesslist.go

```diff
@@ -136,6 +136,11 @@ func (c *Client) DeleteAllAccessLists(ctx context.Context) error {
 	return trace.NotImplemented("DeleteAllAccessLists not supported in the gRPC client")
 }
 
+// CountAccessListMembers will count all access list members.
+func (c *Client) CountAccessListMembers(ctx context.Context, accessListName string) (int, error) {
+	return 0, trace.NotImplemented("CountAccessListMembers not supported in the gRPC client")
+}
+
```

Comment:
nit: I'm not a fan of the pattern of adding fake client methods to satisfy interfaces.  We do it in a handful of places, but IMO it's preferable to follow the pattern of having a separate "internal" interface that implements the extra methods local methods (see `services.StatusInternal` for a concise example).  Reasonable people can and do disagree on this point, but so feel free to leave as-is if you prefer.

Reply:
+1 from me too

Reply:
I agree, but I'm not sure I want to do that as part of this PR. If it's alright with you, I'd like to do this as a follow on to clean this up.

Reply:
For changes to `api`, I think it's better to wait and get things into their final form.  We're supposed to make an effort to keep `api` stable, and while removing a method that never functioned is a fairly benign form of instability, it's still preferable to avoid it unless there is a compelling reason not to.

Reply:
I'll give this a shot.

Reply:
I don't think this is doable because part of the speedup here is caching an internal only function call, `CountAccessListMembers`.

Reply:
To be more specific, somewhere in the stack I won't be able to convert an internal-only representation of an access list client into one that supports caching. In `lib/auth/accesspoint/accesspoint.go`:

```
		SAMLIdPServiceProviders: cfg.Services,
		UserGroups:              cfg.Services,
		Okta:                    cfg.Services.OktaClient(),
		AccessLists:             cfg.Services.AccessListClient(), <-- This does not implement AccessListInternal
		SecReports:              cfg.Services.SecReportsClient(),
		UserLoginStates:         cfg.Services.UserLoginStateClient(),
```

Reply:
Okay, after talking with Forrest, I think the thing to do here is that the client needs to respect these API calls, but only for `SystemRoles` to make sure remote caching can work.

Reply:
I've implemented the count in the client/will implement it in the gRPC service. Let me know what you all think.